### PR TITLE
SO-4007 Restore effectiveTime during RF2 Delta imports

### PR DESCRIPTION
**Pull request by cmark**
This PR adds the ability to run the `effectiveTime` restore process during RF2 Delta imports.
This change will make sure that the `effectiveTime` field remains set to the proper latest released version `effectiveTime` even if an RF2 Delta import contains a row with the same properties but with the `effectiveTime` unset.

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java [hunk 1]
```diff
@@ -80,14 +83,33 @@
...
+	 * @param jobKey - the logical key assigned to the job
+	 * @param predicate - the predicate filter to further customize the search process
+	 * @return
+	 */
+	default boolean isJobRunning(String jobKey, Predicate<RemoteJobEntry> predicate) {
+		checkNotNull(predicate, "Predicate should not be null");
 		// check first if this context is running inside a job with the given jobKey
 		Optional<RemoteJob> job = optionalService(RemoteJob.class);
 		if (job.isPresent() && Objects.equals(jobKey, job.get().getKey())) {
 			return true;
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java]: The predicate is not applied to the current job.

**Review by apeteri — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java (hunk 1 above)
Comment (cmark) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java]: Right, thanks! 😅

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java [hunk 2]
```diff
@@ -80,14 +85,34 @@
...
+	
+	/**
+	 * Returns <code>true</code> if any job present with the given jobKey in {@link RemoteJobEntry#isRunning()} state and matches the given parameters predicate, <code>false</code> otherwise.
+	 * 
+	 * @param jobKey - the logical key assigned to the job
+	 * @param parametersPredice - the predicate to filter the job by its parameters
+	 * @return
+	 */
+	default boolean isJobRunning(String jobKey, Predicate<Map<String, Object>> parametersPredice) {
+		checkNotNull(parametersPredice, "Parameters Predicate should not be null");
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java]:
```suggestion
	 * @param parametersPredicate - the predicate to filter the job by its parameters
	 * @return
	 */
	default boolean isJobRunning(String jobKey, Predicate<Map<String, Object>> parametersPredicate) {
		checkNotNull(parametersPredicate, "Parameters predicate should not be null");
```

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java [hunk 3]
```diff
@@ -80,14 +85,34 @@
...
 		// check first if this context is running inside a job with the given jobKey
 		Optional<RemoteJob> job = optionalService(RemoteJob.class);
-		if (job.isPresent() && Objects.equals(jobKey, job.get().getKey())) {
-			return true;
+		if (job.isPresent()) {
+			return Objects.equals(jobKey, job.get().getKey()) && parametersPredice.test(job.get().getParameters(service(ObjectMapper.class)));
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java]:
```suggestion
			return Objects.equals(jobKey, job.get().getKey()) && parametersPredicate.test(job.get().getParameters(service(ObjectMapper.class)));
```

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java [hunk 4]
```diff
@@ -80,14 +85,34 @@
...
 		// if not inside a job context or running in non-job context check the jobs index
-		return JobRequests.prepareSearch().one().filterByKey(jobKey).build().execute(this).first().map(RemoteJobEntry::isRunning).orElse(false);
+		return JobRequests.prepareSearch().one()
+				.filterByKey(jobKey)
+				.build()
+				.execute(this)
+				.first()
+				.filter(RemoteJobEntry::isRunning)
+				.map(j -> j.getParameters(service(ObjectMapper.class)))
+				.filter(parametersPredice)
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ServiceProvider.java]:
```suggestion
				.filter(parametersPredicate)
```

**Review by apeteri — APPROVED** [review]
Some typos need to be fixed, but otherwise LGTM!

**Comment by cmark** [comment]
Merging!

**Pull request merged**